### PR TITLE
fix(buildkite): Catch '?' yaml syntax for plugins

### PR DESCRIPTION
**Pull request by scemily13**
## Changes

Modify the buildkite dependency line matcher to allow for plugins of the form: 
`- ? ssh://git@github.company.com/some-org/some-plugin.git#v3.2.7`

## Context

Renovate is not opening PRs on repositories that specify buildkite plugins in the above-mentioned form.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/buildkite/extract.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export function extractPackageFile(content: string): PackageFile | null {
     for (const line of lines) {
       // Search each line for plugin names
       const depLineMatch = regEx(
-        /^[\s-]*(?<depName>[^#\s]+)#(?<currentValue>[^:]+)/
+        /^[\s-?]*(?<depName>[^#\s]+)#(?<currentValue>[^:]+)/
```
Comment (viceice) [inline lib/modules/manager/buildkite/extract.ts:16]: seems wrong to me. regex allows ` - ? - ? ...`.
should be more strict.

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/modules/manager/buildkite/extract.ts (hunk 1 above)
Comment (scemily13) [reply lib/modules/manager/buildkite/extract.ts:16]: I can make it more strict!

Comment (jhpierce) [inline lib/modules/manager/buildkite/extract.ts:16]: The `-` character has special meaning when it is between two other characters inside `[]`, so prefer to add it at the beginning or end of the character group

File: lib/modules/manager/buildkite/extract.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ export function extractPackageFile(content: string): PackageFile | null {
     for (const line of lines) {
       // Search each line for plugin names
       const depLineMatch = regEx(
-        /^[\s-]*(?<depName>[^#\s]+)#(?<currentValue>[^:]+)/
+        /^[\s-]*(\?\s)?(?<depName>[^#\s]+)#(?<currentValue>[^:]+)/
```
Comment (viceice) [inline lib/modules/manager/buildkite/extract.ts:16]:
```suggestion
        /^\s*(?:-\s+(?:\?\s+)?)?(?<depName>[^#\s]+)#(?<currentValue>[^:]+)/
```
I'm not sure, but is the dash optional for those plugins?

Comment (scemily13) [reply lib/modules/manager/buildkite/extract.ts]: It is optional, and this suggestion works perfectly!

**Review by viceice — APPROVED** [review]

**Comment by rarkins** [comment]
Thank you for the fix, @scemily13

**Pull request merged**